utils.go

```diff
@@ -16,7 +16,15 @@ func newHTTPClient(u *url.URL, tlsConfig *tls.Config, timeout time.Duration) *ht
 	switch u.Scheme {
 	default:
 		httpTransport.Dial = func(proto, addr string) (net.Conn, error) {
-			return net.DialTimeout(proto, addr, timeout)
+			conn, err := net.DialTimeout(proto, addr, timeout)
+			if tcpConn, ok := conn.(*net.TCPConn); ok {
+				// Set TCP user timeout. Sender breaks TCP connection
+				// if packets are not acknowledged after 20 seconds. This is a
+				// relatively new TCP option to improve dead peer detection.
+				// Do not fail newHTTPClient if OS doesn's support it.
+				SetTCPUserTimeout(tcpConn, 20*time.Second)
```

Comment:
@vieux It's an option. The existing `timeout` is for TCP connection setup. I was thinking several options including `timeout, 20s, 2 * timeout, timeout + 10s, max(timeout, 20s)`. I hesitated with `timeout` because I'm afraid of aggressive user setting on `timeout`. Dead peer detection shouldn't be too aggressive.